### PR TITLE
TigonSQL dependent on tigon-common and tigon-api; Use File instead of Location.

### PR DESCRIPTION
**Pull request by gokulavasan**
- Move classes dependent on Hadoop from tigon-common to tigon-core. Now tigon-sql is dependent only on tigon-common (and tigon-api).
- Use File instead of Location in tigon-sql since all operations are on local filesystem
- Cleaned up pom files because of the move above and some housekeeping

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -25,7 +25,10 @@ See [Surefire doc](http://maven.apache.org/surefire/maven-surefire-plugin/exampl
     mvn package -DskipTests -pl tigon-examples -am -amd -P examples
 
 ### Build Tigon Distribution
-    mvn package -DskipTests -pl tigon-examples -am -amd -P examples && mvn clean package -DskipTests -Pdist
+    mvn clean package -DskipTests -P examples -pl tigon-examples -am -amd && mvn package -DskipTests -Pdist
```
Comment (sreevatsanraman) [inline BUILD.md:28]: we don't need -P examples and -pl tigon-examples. only -P examples will do.

Comment (sreevatsanraman) [inline BUILD.md:28]: Also we don't need the second mvn package. We can add both the profiles in the first command -P examples,dist

File: tigon-sql/src/main/java/co/cask/tigon/sql/internal/StreamBinaryGenerator.java [hunk 2]
```diff
@@ -47,26 +45,26 @@
...
-  public StreamBinaryGenerator(Location dir, InputFlowletSpecification spec) {
+  public StreamBinaryGenerator(File dir, InputFlowletSpecification spec) {
     this.dir = dir;
     this.spec = spec;
   }
 
-  public Location createStreamProcesses() {
-    Location configDir = null;
+  public File createStreamProcesses() {
+    File configDir = null;
```
Comment (sreevatsanraman) [inline tigon-sql/src/main/java/co/cask/tigon/sql/internal/StreamBinaryGenerator.java]: Move this inside try block you don't have to initialize to null.

**Comment by sreevatsanraman** [comment]
Minor comments. Fix and merge

File: BUILD.md (hunk 1 above)
Comment (gokulavasan) [inline BUILD.md:28]: so will the command below be enough?
`mvn clean package -DskipTests -Pexamples,dist`

Comment (gokulavasan) [inline BUILD.md:28]: Looks like we need the command in the current form. I will merge after build passes.

**Comment by gokulavasan** [comment]
Build passed. Merging now.

**Pull request merged**